File: examples/raw-python/hmac_lookup_transaction.py
from urllib.parse import quote
from payxcommerce import DEFAULT_BASE_URL, endpoint, hmac_headers, json_request, print_response

PUBLIC_KEY = "YOUR_PUBLIC_KEY"
SECRET_KEY = "YOUR_SECRET_KEY"
TRANSACTION_REFERENCE = "PXTRX-YYYYMMDD-XXXXXX"

path = "/transactions/" + quote(TRANSACTION_REFERENCE, safe="")
print_response(json_request("GET", endpoint(DEFAULT_BASE_URL, path), hmac_headers(PUBLIC_KEY, SECRET_KEY, "")))

File: examples/raw-nodejs/hmac-lookup-transaction.js
'use strict';

const { DEFAULT_BASE_URL, endpoint, hmacHeaders, jsonRequest, printResponse } = require('./payxcommerce');

const publicKey = 'YOUR_PUBLIC_KEY';
const secretKey = 'YOUR_SECRET_KEY';
const transactionReference = 'PXTRX-YYYYMMDD-XXXXXX';

jsonRequest('GET', endpoint(DEFAULT_BASE_URL, `/transactions/${encodeURIComponent(transactionReference)}`), hmacHeaders(publicKey, secretKey, '')).then(printResponse).catch((error) => {
  console.error(error.message);
  process.exit(1);
});
